workflow-templates/im-test-k6-manual-advanced.yml

```diff
@@ -0,0 +1,66 @@
+name: Manual trigger K6 test - Advanced
+
+on: 
+  workflow_dispatch:
+    inputs:
+      envToTest:
+        description: 'Environment to run test against [[1] dev, [2] qa, [3] stage]'   
+        required: true
+        default: '1'
+      testFolderRoot:
+        description: 'Test folder root directory from root of repo'
+        required: true
+        default: "tests/k6"
+      testFile:
+        description: 'K6 test file to run'     
+        required: true
+      k6_extra_params:
+        description: 'K6 command extra paramaters'
+        required: false
+        default: '' # You can set default paramaters to run on every K6 test.
+
+jobs:
+  k6_test:
+    runs-on: [self-hosted, ubuntu-20.04]
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+
+    - name: Setup K6
+      uses: im-open/setup-k6-perf-testing@v1.0.0
+      with:
+        version: 0.33.0
+
+    - name: Validate test file exists
+      id: validation
+      run: |
+        k6_env='dev'
+        if [[ "${{ github.event.inputs.envToTest }}" == "1" ]]; then
+          echo "env: dev"
+        elif [[ "${{ github.event.inputs.envToTest }}" == "2" ]]; then
+          echo "env: qa"
+          k6_env='qa'
+        elif [[ "${{ github.event.inputs.envToTest }}" == "3" ]]; then
+          echo "env: stage"
+          k6_env='stage'
+        else
+          echo "Invalid option"
+          exit 1
+        fi
```

Comment:
This whole thing should be replaced with https://github.com/im-open/map-input-action/pull/3.  Each of the places where this is done should be updated as well.

Reply:
Added as requested